src/vs/workbench/services/editor/common/customEditorLabelService.ts

```diff
@@ -115,7 +119,18 @@ export class CustomEditorLabelService extends Disposable implements ICustomEdito
 		if (!this.enabled) {
```

Comment:
Let's not fill the cache if the user is not using this feature (when he doesn't have any patterns). You can add `this.patterns.length === 0` to the `!this.enabled` check and remove the pattern length check in `applyPatterns()`